ocaml/typing/typecore.ml

```diff
@@ -7362,7 +7445,7 @@ and type_option_some env expected_mode sarg ty ty0 =
 
 (* [expected_mode] is the expected mode of the field. It's already adjusted for
    allocation, mutation and modalities. *)
-and type_label_exp create env (arg_mode : expected_mode) loc ty_expected
+and type_label_exp ?(overwrite = No_overwrite) create env (arg_mode : expected_mode) loc ty_expected
```

Comment:
Would you want `overwrite` to be an optional argument on `type_expect_`? There I made this choice quite explicitly: I don't want anyone to pass an `overwrite` unless they are in a context where overwrites can happen. For consistency, I made the same choice here. However, I do agree that `type_label_exp` is the kind of function where overwrite will often happen, so perhaps it should actually be a non-optional argument. What do you think?

Reply:
I think the reason to make an argument optional is if there are a lot of calls, and it would be clutterful to require all of those calls to change. That's a little different than the motivation you suggest, because I think it's nice when reading to see e.g. `~overwrite:No_overwrite` to explicitly signal that no overwriting can happen there.

That said, I'm fine with `type_expect_` (and friends) to use optional arguments here, because of the number of calls.